spring-cloud-dataflow-server-core/pom.xml

```diff
@@ -119,6 +119,12 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+		<dependency>
```

Comment:
We want the following:
1. h2 to be optional
2. h2 to be available during test.

The first thought would be this: 
```xml
<dependency>
  <groupId>com.h2database</groupId>
    <artifactId>h2</artifactId>
    <optional>true</optional>
</dependency>

<dependency>
  <groupId>com.h2database</groupId>
  <artifactId>h2</artifactId>
  <scope>test</scope>
</dependency>
```
That does not work as maven does not like the 2x declaration of the dependency. 

This does not work either (the scope only of test immediately limits it to only test availability)
```xml
<dependency>
  <groupId>com.h2database</groupId>
  <artifactId>h2</artifactId>
  <optional>true</optional>
  <scope>test</scope>
</dependency>
```
Where we landed on optional=true and scope=provided DOES work because 
> [^1] The provided dependencies are available only at compile time and in the test classpath of the project. These dependencies are also not transitive.

[^1]: https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope